### ISSUE TITLE
Failures in test_InsertRetrieveDateTimeTypeColumn

### ISSUE THREAD

**Issue by davidmmooney** [ibmdb/python-ibmdb#513]
(Orignally posted this as a comment to #492 -- I hadn't noticed that the issue was closed. Sorry.)

I see that some of the latest CI runs have been failing in test_InsertRetrieveDateTimeTypeColumn.py with this error:
```
======================================================================

FAIL: test_InsertRetrieveDateTimeTypeColumn (test_InsertRetrieveDateTimeTypeColumn.IbmDbTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_InsertRetrieveDateTimeTypeColumn.py", line 18, in test_InsertRetrieveDateTimeTypeColumn
    obj.assert_expectf(self.run_test_InsertRetrieveDateTimeTypeColumn)
  File "/home/travis/build/kadler/python-ibmdb/IBM_DB/ibm_db/testfunctions.py", line 109, in assert_expectf
    self.assertNotEqual(result, None)
AssertionError: None == None
```

I think the error is here:

    statement = "INSERT INTO tab_datetime (col1, col2, col3) values ('24:00:00', ?, ?)"

Db2 considers "24:00:00" to be a valid time, but Python does not.

Try this patch:
```
diff --git a/IBM_DB/ibm_db/ibm_db_tests/test_InsertRetrieveDateTimeTypeColumn.py b/IBM_DB/ibm_db/ibm_db_tests/test_InsertRetrieveDateTimeTypeColumn.py
index 1862b2e..bce1d53 100644
--- a/IBM_DB/ibm_db/ibm_db_tests/test_InsertRetrieveDateTimeTypeColumn.py
+++ b/IBM_DB/ibm_db/ibm_db_tests/test_InsertRetrieveDateTimeTypeColumn.py
@@ -62,12 +62,15 @@ class IbmDbTestCase(unittest.TestCase):
             rc = ibm_db.execute(stmt)
             result = ibm_db.fetch_row(stmt)
             while ( result ):
-                row0 = ibm_db.result(stmt, 0)
-                row1 = ibm_db.result(stmt, 1)
-                row2 = ibm_db.result(stmt, 2)
-                print(type(row0), row0)
-                print(type(row1), row1)
-                print(type(row2), row2)
+                try:
+                    row0 = ibm_db.result(stmt, 0)
+                    row1 = ibm_db.result(stmt, 1)
+                    row2 = ibm_db.result(stmt, 2)
+                    print(type(row0), row0)
+                    print(type(row1), row1)
+                    print(type(row2), row2)
+                except ValueError as e:
+                    print("Failed to parse row:", e, file=sys.stderr)
                 result = ibm_db.fetch_row(stmt)
 
             ibm_db.close(conn)
```
and for me, at least, it prints
```
test_InsertRetrieveDateTimeTypeColumn (test_InsertRetrieveDateTimeTypeColumn.IbmDbTestCase) ... Failed to parse row: hour must be in 0..23
FAIL
```
Changing the time in the INSERT statement to '00:00:00' allows the test to pass.

(The problem with this theory is that all the SQL->Python date conversion code in ibm_db.c appears to do a proper `% 24` on the hour values before passing them down to the Python layer...)

**Comment by SabaKauser** [comment]
can you please open a PR. That way , travis CI will tell me if I can merge them directly than me having to add these..  Just trying to save some time :P

**Comment by davidmmooney** [comment]
This is my error. I hadn't noticed that the package at https://pypi.org/project/ibm-db/ is over a year old, so when I ran the test against the default `pip install ibm_db` package, I didn't pick up the fixes for this exact problem delivered in December. When I installed the package from source (`pip install .`) and re-ran the test it was clean.

I'll close this issue, but might I suggest that you push a new version of ibm_db to pypi?
